docs/shared/concepts/sync-generators.md

```diff
@@ -0,0 +1,46 @@
+# Sync Generators
+
+In Nx 19.7, you can use sync generators to keep applicable files in sync with your project graph. These can be for global configuration files or scripts, or at the task level to ensure to that project level configuration files are in sync before a task is run.
```

Comment:
A couple of clarifications here:

- Sync generators "can" keep files in sync with the project graph, but they are not limited to that. They can do anything and are not restricted to using the project graph as the source of truth.
- Similarly, sync generators "can" update configuration files but are not restricted to that type of files. They could also add/update/delete any file.

The example below about formatting files is a clear example of the above, where the project graph is not the source of truth and any file could be formatted.